### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by grummbeer**
- add phpstan
- add phpstan to workflow
- enable workflow checks for pull_request
- add tests with coverage
- fix skipped test #22 (partially)
- cleanup
  - php 8 features
  - type hinting
- no functional changes
- some tests still missing

### CODE REVIEW

**Comment by grummbeer** [comment]
@amenk add composer.lock?

**Comment by amenk** [comment]
> @amenk add composer.lock?

no sure - https://getcomposer.org/doc/02-libraries.md#lock-file does not have a final recommendation, do you see benefits in adding it?

> Please do some real usage tests.

We are using it with the sister project https://github.com/Mestrona/mbank
Probably that needs to be upgraded as well, if I find the time. Then I can test it.

Did you do some tests already? What is your usecase?

**Comment by grummbeer** [comment]
> composer.lock

OK, i see. No need for it. 

I haven't tested all commands yet. I have not made any functional changes, just a cleanup. So everything _should_ behave as before. Your mbank works, only this line needs to be changed to the newer money api `getCurrency()->getCode()`.

https://github.com/Mestrona/mbank/blob/2f175ffb3ebfa0884596f874bb58478b0e213d2f/app/BankTransactionTable.php#L41

My use case is similar to mbank, just with more frontend.

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -0,0 +1,46 @@
+parameters:
```
Comment (amenk) [inline phpstan-baseline.neon:1]: would be cool to get the baseline empty though ;-)

**Review by amenk — APPROVED** [review]

**Comment by amenk** [comment]
Looks good, I guess I can merge - shall I tag a new release (5.0.0) as well or do you have further plans?

File: phpstan-baseline.neon (hunk 1 above)
Comment (grummbeer) [reply phpstan-baseline.neon:1]: Yeah would be great. Some of it from UserMatcher/AccountMatcher are removed now. The others are a nice future improvement.

**Comment by grummbeer** [comment]
> plans

- some modernizing eg. 
   - "preg_match(/simple matching/) > str_contains"
   - "false !== str_pos() > str_contains"
   - "$nullableObject->get() > $nullableObject?->get()" using null-safe-operator

- bugs
  - SetAppropriateITanModeCommand don't work, unsupported Parameter "--bank"  and "--user" should be uniqueId and not userId

Would do this in separate MRs, this one can be merged.

File: phpstan-baseline.neon (hunk 1 above)
Comment (grummbeer) [reply phpstan-baseline.neon:1]: Add types for parameters in ContextXMLRenderer, removed from baseline

**Comment by grummbeer** [comment]
Will not send more commits to this branch :-) Ready to merge from my side.

**Comment by amenk** [comment]
Release is tagged

https://github.com/Mestrona/aqbanking-php/releases/tag/4.0.0

Feel free to open issues for the todos,

Thank you so much!

**Pull request merged**